e2e_test/batch/functions/substr.slt.part

```diff
@@ -0,0 +1,9 @@
+# issue: https://github.com/risingwavelabs/risingwave/issues/7632
+statement error out of range
+select substr('W7Jc3Vyufj', (INT '-2147483648'));
```

Comment:
Actually PostgreSQL returns the whole string for this. It does not report NumericOutOfRange when start+count overflows as well.

https://github.com/postgres/postgres/blob/REL_15_1/src/backend/utils/adt/varlena.c#L813